langfun/core/llms/llama_cpp.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2023 The Langfun Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Language models from llama.cpp."""
+
+from typing import Annotated
+
+import requests
+
+import langfun.core as lf
+
+@lf.use_init_args(['url'])
+class RemoteLLaMACPP(lf.LanguageModel):
+  """The remote LLaMA C++ model.
+
+  The Remote LLaMA C++ models can be launched via
+  https://github.com/ggerganov/llama.cpp/tree/master/examples/server
+  """
+
+  url: Annotated[
+      str | None,
+      'The name of the model to use.',
+  ] = None
+
+  name: Annotated[
```

Comment:
Is the model identified by the url alone? Thus this is not needed?

Reply:
Any user can launch any kind of model with whatever url as he wants. The name is not associated with the url, I view it as a model name to indicate the real model behind the url for the convenience purpose. WDYT?

Reply:
It makes sense. Thanks Xuanyi.